adr/0004-supported-python-versions.md

```diff
@@ -0,0 +1,36 @@
+# Supported Python Versions
+
+Date: 2017-11-20
+
+## Status
+
+Proposed
+
+## Context 
+
+eLife has numerous projects written completely and partly with the [Python programming language](https://www.python.org/).
+
+In order to provide language version consistency across projects we need to get a consensus on which versions we are going to support. 
+
+## Decision
+
+We will use Python 2.7.14 as our default version for any project that solely uses or supports Python 2.
+
+We will use Python 3.6.3 as our default supported version for any project that solely uses or supports Python 3.
```

Comment:
Do we want patch versions here?

Reply:
Probably not, was just unsure how granular we were looking to go. I have certainly had to consider it in the past due to dependancy bugs with certain Python patch versions `Pillow` comes to mind, but perhaps its not required here.

Reply:
For `2.7.14` yes, or `>=2.7.14` instead but I don't know if more versions will ever come out there.

Reply:
So `latest Python 2.7` and `latest Python 3.6`?

Reply:
Don't know about 3.6, but `2.7.14` was a minimal version in the sense that it's what we have been testing with to solve a few issues like SNI support that weren't working in the default 2.7. If a newer 2.7.x one is available, it would probably work, but it's not strictly necessary to upgrade.